packages/cli/src/Server.ts

```diff
@@ -167,6 +168,23 @@ export class Server extends AbstractServer {
 		this.webhooksEnabled = !config.getEnv('endpoints.disableProductionWebhooksOnMainProcess');
 	}
 
+	async init(): Promise<void> {
+		if (!config.getEnv('userManagement.jwtSecret')) {
+			// If we don't have a JWT secret set, generate
+			// one based and save to config.
+			const { encryptionKey } = Container.get(InstanceSettings);
+
+			// For a key off every other letter from encryption key
+			// CAREFUL: do not change this or it breaks all existing tokens.
+			let baseKey = '';
+			for (let i = 0; i < encryptionKey.length; i += 2) {
+				baseKey += encryptionKey[i];
+			}
+			config.set('userManagement.jwtSecret', createHash('sha256').update(baseKey).digest('hex'));
+		}
```

Comment:
Would it be possible to write a test for this?

Also maybe it would make sense to add a comment that jwtSecret caches the secret and it needs to be generated before that

Reply:
I tried adding a test to assert that `JWTService` is initialized after `config.set('userManagement.jwtSecret'` is called, but due to how `Server.ts` is, it's not easy to mock everything, and the test ends up being rather slow.
Maybe we should move this code to `JWTService`, and make sure only use this service for any JWT specific code.

Reply:
I think that would make sense. At least then the code to generate the key is close to where it is used as well